README.md

```diff
@@ -29,7 +37,484 @@ The library design is inspired by the work in [Exploring HTN Planners through ex
+
+```typescript
+// If successful, it should result in the following state: {counter: 3, needsWrite: true}
+agent.seek({ counter: 3, lastRead: UNDEFINED });
+
+// If successful, it should result in the following state: {counter: 3, needsWrite: true, lastRead: undefined}
+agent.seek({ counter: 3, lastRead: undefined });
+```
+
+We'll learn later how we can add an `op` property to tasks to tell Mahler when a task is applicable to a `delete` operation.
+
+One last thing before moving on from this topic. What if you assign a required value the value of `UNDEFINED`?
+/inv
+
+```typescript
+// Compilation fais with: `type symbol is not assignable to number|undefined`
+agent.seek({ counter: UNDEFINED });
+```
+
+The compiler will prevent you from doing that to avoid ending up with inconsistent state (as long as you don't use `as any`).
+
+## Methods
+
+Now, as programmers, we want to be able to build code by composing simpler behaviors into more complex ones. We might want to guide the planner towards a specific solution, using the primitives we already have. For instance, let's say we want to help the planner get to a solution faster as increasing one by one takes too much time. We want to define a `plusTwo` task, that increases the counter by `2`. We could create another primitive task to update the counter by two, but as programmers, we would like to reuse the code we have already defined. We can do that using methods.
+
+```typescript
+// We'll go back to the simpler definition for this example
+const plusOne = Task.from({
+	condition: (state: number, { target }) => state < target,
+	effect: (state: number) => state + 1,
+	description: '+1',
+});
+
+const plusTwo = Task.from({
+	// We want this method to be chosen only if the difference between the current
+	// state and the target is bigger than one
+	condition: (state: number, { target }) => target - state > 1,
+	// Defining the method property makes this task into a method
+	method: (_: number, { target }) => [plusOne({ target }), plusOne({ target })],
+	description: '+2',
+});
+```
+
+Now there is a lot happening here. We have replaced `effect` and `action` in the task constructor with method. We are also directly using the `plusOne` task as a function and passing the target as one of the properties of the argument object. The code above is the way to create composite tasks in `Mahler`, called methods. Using `method` tells the task constructor to return a `MethodTask` instead of an `ActionTask`. Objects generated by the task constructor are callable, and receive part of the `Context` as the argument. Passing the context to a task _grounds_ the task, and this happens normally as part of the planning process. However, we reuse this mechanism to be able to reuse tasks as part of methods. In fact, the return value is also callable.
```

Comment:
> Passing the context to a task _grounds_ the task, and this happens normally as part of the planning process.

What does "ground" mean in this context? From what I get here, we are creating an "instance" of the task bound to the passed context, but I am not sure I am getting the whole idea. A link here would probably be nice, as "ground" is not very Googleable.

Reply:
Hmm, perhaps using "grounding" is a bit confusing. "Binding the task" to the context as you put it sounds more intuitive. I use grounding in the sense of https://en.wikipedia.org/wiki/Ground_expression, but that might not be the right usage.